### PR TITLE
Desc of anonymous variables

### LINKED ISSUE #230: Documentation: Adding how to define anonymous variables
At https://wikunia.github.io/ConstraintSolver.jl/stable/supported/#Supported-variables it would be great if it also describe how to define anonymous variables. 

Anonymous variables are - what I understand - needed if one plan to use them in a separate function that is used more than once in a model, otherwise a name clash error occurs: `An object of name b is already attached to this model. If this is intended, consider using the anonymous construction syntax, e.g., x = @variable(model, [1:N], ...) where the name of the object does not appear inside the macro.`

E.g.
```julia
   y = @variable(model, [1:n], CS.Integers(1:n) # array of length n
   b = @variable(model, [1:1], Bin)  # single variable
```

Note: For single anonymous variable it seems that the `[1:1]` construct is needed.

### PR DESCRIPTION
**Pull request by Wikunia**
Fixes #230 

@hakank maybe you can have a short review of this. Do I miss something that you've encountered to be useful?

### CODE REVIEW

File: docs/src/supported.md [hunk 1]
```diff
@@ -31,6 +31,23 @@ Additionally you can specify a set of allowed integers:
 @variable(m, x, CS.Integers([1,3,5,7]))
 ```
 
+### Anonymous variables
+
+Besides the named way of defining variables it's also possible to have anonymous variables as provided by [JuMP.jl](https://github.com/jump-dev/JuMP.jl). 
+
+This can be useful when one needs to create temporary variables for reformulations of the problem. The values of these variables can't be easily accessed later on but they avoid the problem of needing a new name for all temporary variables.
```
Comment (hakank) [inline docs/src/supported.md:38]: Perhaps also note that if the anonymous variables are returned from a function (constraint) then they can be used as a variables in the main model, e.g. for printing/debugging.

**Review by hakank — CHANGES_REQUESTED** [review]
Other than my comment about returning anonymous variables, I think it's great!

**Comment by Wikunia** [comment]
The new update gives an idea of basic usage which can be improved but might be helpful. I'll improve this further tmr.

File: docs/src/supported.md (hunk 1 above)
Comment (Wikunia) [reply docs/src/supported.md]: Do you mind having another look?

Comment (hakank) [reply docs/src/supported.md]: Yes, Great with a complete (although a bit contrived) model. I say go!

Comment (Wikunia) [reply docs/src/supported.md]: Haha yeah was the simplest even though kinda useless model I could think off. 
It can always be improved :smile:

**Pull request merged**